src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -1324,6 +1324,20 @@ trait Implicits {
       }
     }
 
+    /** Creates a tree will produce a ValueOf instance for the requested type.
+      * An EmptyTree is returned if materialization fails.
+      */
+    private def valueOfType(tp: Type): SearchResult = {
+      def success(t: Tree) = wrapResult(Apply(Select(New(gen.scalaDot(tpnme.ValueOf)), nme.CONSTRUCTOR), List(t)))
```

Comment:
The change you suggest here generates,
```
Apply(Select(This(TypeName("ValueOf")), termNames.CONSTRUCTOR), List(Literal(Constant(...))))
```
rather than,
```
Apply(Select(New(Select(Ident(scala), TypeName("ValueOf"))), termNames.CONSTRUCTOR), List(Literal(Constant(...))))
```
which is what the original generates. The new version fails, I think because the type argument isn't inferred correctly. I'll stick with what I had before for now.